railties/test/backtrace_cleaner_test.rb

```diff
@@ -1,24 +1,37 @@
 require 'abstract_unit'
 require 'rails/backtrace_cleaner'
 
-class BacktraceCleanerVendorGemTest < ActiveSupport::TestCase
+class BacktraceCleanerTest < ActiveSupport::TestCase
   def setup
     @cleaner = Rails::BacktraceCleaner.new
   end
 
   test "should format installed gems correctly" do
-    @backtrace = [ "#{Gem.path[0]}/gems/nosuchgem-1.2.3/lib/foo.rb" ]
-    @result = @cleaner.clean(@backtrace, :all)
-    assert_equal "nosuchgem (1.2.3) lib/foo.rb", @result[0]
+    backtrace = [ "#{Gem.path[0]}/gems/nosuchgem-1.2.3/lib/foo.rb" ]
+    result = @cleaner.clean(backtrace, :all)
+    assert_equal "nosuchgem (1.2.3) lib/foo.rb", result[0]
   end
 
   test "should format installed gems not in Gem.default_dir correctly" do
-    @target_dir = Gem.path.detect { |p| p != Gem.default_dir }
+    target_dir = Gem.path.detect { |p| p != Gem.default_dir }
     # skip this test if default_dir is the only directory on Gem.path
     if @target_dir
-      @backtrace = [ "#{@target_dir}/gems/nosuchgem-1.2.3/lib/foo.rb" ]
-      @result = @cleaner.clean(@backtrace, :all)
-      assert_equal "nosuchgem (1.2.3) lib/foo.rb", @result[0]
+      backtrace = [ "#{target_dir}/gems/nosuchgem-1.2.3/lib/foo.rb" ]
+      result = @cleaner.clean(backtrace, :all)
+      assert_equal "nosuchgem (1.2.3) lib/foo.rb", result[0]
     end
   end
+
+  test "should consider traces from irb lines as User code" do
+    backtrace = [ "from (irb):1",
+                  "from /Path/to/rails/railties/lib/rails/commands/console.rb:77:in `start'",
+                  "from /Path/to/rails/railties/lib/rails/commands/console_helper.rb:9:in `start'",
+                  "from /Path/to/rails/railties/lib/rails/commands/commands_tasks.rb:78:in `console'",
+                  "from /Path/to/rails/railties/lib/rails/commands/commands_tasks.rb:49:in `run_command!'",
+                  "from /Path/to/rails/railties/lib/rails/commands.rb:18:in `<top (required)>'",
+                  "from bin/rails:4:in `require'",
+                  "from bin/rails:4:in `<main>'" ]
```

Comment:
That's a lot of array, given the following assertion is only checking that the first entry survives :smile: